### ISSUE TITLE
Docs build fails due to warnings after sphinx update

### ISSUE THREAD

**Issue by oscarbenjamin** [sympy/sympy#19551]
See #19550 and https://travis-ci.org/github/sympy/sympy/jobs/698127486

The docs build is failing for master on Travis because of warnings. The warnings are those from #19316. It seems that an update of sphinx from 3.1.0 to 3.1.1 results in these warnings appearing while building the docs. An example is
```
WARNING: autodoc: failed to import function 'logic.boolalg.SOPform' from module 'sympy'; the following exception was raised:
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.8.0/lib/python3.8/site-packages/sphinx/util/inspect.py", line 324, in safe_getattr
    return getattr(obj, name, *defargs)
  File "/home/travis/build/sympy/sympy/sympy/__init__.py", line 677, in __getattr__
    return getattr(self.mod, name)
AttributeError: module 'sympy.core.logic' has no attribute 'boolalg'
```
What has happened I think is that sphinx is trying to get module boolalg by doing
```
import sympy
sympy_logic = getattr(sympy, 'logic')
sympy_logic_boolalg = getattr(sympy_logic, 'boolalg')
```
rather than
```
from sympy.logic import boolalg
```
Using `getattr` like this means that `sympy_logic` ends up being the `sympy.core.logic` module rather than the `sympy.logic` package. The `sympy.core.logic` module is available as `getattr(sympy, 'logic')` but wrapped in a deprecation warning since #19316.

The warnings and the attributes on the main sympy module were added in #19316 to warn anyone who was importing the modules that were previously importable by accident. Those warnings have gone out in the 1.6 release. I think that we can just remove them and basically revert #19316 so that the accidental imports become `ImportError` rather than a warning.

**Comment by oscarbenjamin** [comment]
I've opened an issue with sphinx (no response yet):
https://github.com/sphinx-doc/sphinx/issues/7841

**Comment by asmeurer** [comment]
To clarify we are currently pinning Sphinx to 3.1.0 in Travis to workaround this. So we either need an upstream fix or a workaround that works with the latest Sphinx version, so that we don't have to pin any more. If anyone comes across the above error message when building the docs, you will need to downgrade your Sphinx to 3.1.0 to build the SymPy docs.

**Comment by oscarbenjamin** [comment]
There is a suggestion from @tk0miya in https://github.com/sphinx-doc/sphinx/issues/7841#issuecomment-644883102

It seems that we could change e.g. autofunction references to use `::` to separate the module name from the object name. For example I did this in `solveset.rst` and it seems to fix the warnings related to the `solveset` module:
```diff
diff --git a/doc/src/modules/solvers/solveset.rst b/doc/src/modules/solvers/solveset.rst
index d0ff4c2a40..872c541c7c 100644
--- a/doc/src/modules/solvers/solveset.rst
+++ b/doc/src/modules/solvers/solveset.rst
@@ -562,19 +562,19 @@ Or one may manually rewrite the equation as an expression equal to 0::
 The first argument for :func:`solveset` is an expression (equal to zero) or an equation and the second argument
 is the symbol that we want to solve the equation for.
 
-.. autofunction:: sympy.solvers.solveset.solveset
+.. autofunction:: sympy.solvers.solveset::solveset
 
-.. autofunction:: sympy.solvers.solveset.solveset_real
+.. autofunction:: sympy.solvers.solveset::solveset_real
 
-.. autofunction:: sympy.solvers.solveset.solveset_complex
+.. autofunction:: sympy.solvers.solveset::solveset_complex
 
-.. autofunction:: sympy.solvers.solveset.invert_real
+.. autofunction:: sympy.solvers.solveset::invert_real
 
-.. autofunction:: sympy.solvers.solveset.invert_complex
+.. autofunction:: sympy.solvers.solveset::invert_complex
 
-.. autofunction:: sympy.solvers.solveset.domain_check
+.. autofunction:: sympy.solvers.solveset::domain_check
 
-.. autofunction:: sympy.solvers.solveset.solvify
+.. autofunction:: sympy.solvers.solveset::solvify
 
 linear_eq_to_matrix
 -------------------
```
We should probably apply that kind of change everywhere and then update the style guide.

I guess that there's a way to do this quickly with sed. There are lots of examples:
```console
$ git grep autofunction | grep sympy. | wc -l
     332
$ git grep autoclass | grep sympy. | wc -l
     230
```

**Comment by oscarbenjamin** [comment]
I guess a fix for this should probably go into 1.6.1 as well as master.

**Comment by oscarbenjamin** [comment]
This is now fixed on master and on the 1.6 branch. I'll close this but and open a new issue to discuss updating the style guide and whether we want to apply this more widely.
